CONTRIBUTING.md

```diff
@@ -71,24 +72,81 @@ git status
+If you are preparing your first pull request on this branch, rebase to master
 
 ```sh
-git push -u origin my-great-stuff
+git checkout master
+git pull https://github.com/decred/dcrdata.git
+git checkout my-great-stuff
+git rebase -i master
 ```
 
-7. Create the pull request
+In the text editor, change the command from `pick` to `fixup` or `squash` for **all but the top** commit. Use `fixup` for little touchups to discard the commit's log message. If you want to use a different
+commit message for everything, change the command from  `pick` to `reword` on the top commit.
+It should look something like this before saving.
+
+
+![alt text](https://i.imgur.com/fOtaYtb.png "Rebase commmit command guide")
+
+
+7. **If you have conflicts**, you will need to resolve them by iterating through the diffs one conflicting commit at a time
+
+```sh
+# resolve conflicts
+git add file1.go, file2.go, ...
+git rebase --continue
+# repeat until rebase completes
+```
+
+8. Push your commit to GitHub
+
+Assuming `myremote` is the name of the remote used for *your* repository:
+
+```sh
+git push -u myremote my-great-stuff
+```
+
+9. Create the pull request
+
+On Github, select your branch in the dropdown menu (1) and click (2) to start a pull request.
 
-At the [main dcrdata repo page](https://github.com/decred/dcrdata) (not your fork) you may find a button suggesting that you might want to create a PR from a branch you just pushed. "Compare & pull request" it will say on a big green button. Click it and it will start form to open a new pull request. Always:
+![alt text](https://i.imgur.com/GXZTyiq.png "Pull Request submission guide")
+
+Always:
 
 - Type a detailed comment for the changes you are proposing.  Include motivation and a description of the code change.
 - Highlight any breaking changes.  This includes any syntax changes, added or removed struct fields, interface changes, file renames or deletions, etc.
 - Scroll down and review the code diffs. Verify that the changes are what you expect to see based on your earlier review of the diffs and your git commit log (you did that, right?).
 
 Excellent [PR guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#best-practices-for-faster-reviews) from Kubernetes project.
 
+10. Receive feedback and make changes
+
+You will typically receive feedback from other developers. Before responding, take a moment to review the 
+[Code of Conduct](https://github.com/decred/dcrdata/blob/master/CODE_OF_CONDUCT.md). 
+
+Work through the comments and resolve any confusion with others and then implement the changes necessary.
+
+11. Resubmitting
+
+Similar to step #6 above, you will want to clean up your commits before creating the pull request. 
+As long as it hasn't been a very long time, you don't need to rebase to master. Instead, rebase 
+to the commit of the last pull request. To see a list of commits, you can use

```

Comment:
I don't follow what is meant by "rebase to the commit of the last pull request".  Why not rebase on master?

Reply:
I guess I'm trying to capture our conversation [here](https://github.com/decred/dcrdata/pull/772#issuecomment-433962404). I'm still trying to get used to all the git terminology, so forgive me if these next words are nonsense, but I think what I'm trying to say is that you can clean up little fixups, but don't squash down to master as before.

Reply:
OK, so to do an interactive rebase just to squash the PR branch, but not rebase on master, I'd generally do one of two things:
1. specify how many commits to squash via `git rebase -i HEAD~3` (to squash 3 commits).
2. rebase the entire PR branch back to the *common ancestor* with master (where we branched to make this PR): checkout master and pull latest, then checkout PR branch and then `git rebase -i $(git merge-base master HEAD)`

Reply:
There are [a number of other ways to squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) that can be appropriate too, but if the route is interactive rebase, the above is what I would do.

Reply:
Anyway, it's what you described, but I'm not sure about the sentence "	Find the hash of the commit of your last pull request".  I would specify a number of commits back from head, or the common ancestor with an up-to-date master.

Reply:
> `git rebase -i $(git merge-base master HEAD)`

Since the branch has already been cleaned up in step 1, is the `-i` needed?